central/networkbaseline/manager/manager_impl.go

```diff
@@ -919,6 +920,123 @@ func (m *manager) CreateNetworkBaseline(deploymentID string) error {
+	}
+
+	entities, err := m.getEntitiesByQuery(ctx, deployment.GetClusterId(), query)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get matching entities")
+	}
+
+	entitiesMap := make(map[string]*storage.NetworkEntityInfo)
+	entityFilter := set.NewStringSet()
+	for _, entity := range entities {
+		info := entity.GetInfo()
+		entityFilter.Add(info.GetId())
+
+		// store enriched entities data for lookup later
+		entitiesMap[info.GetId()] = info
+	}
+
+	pred := func(props *storage.NetworkFlowProperties) bool {
+		src := props.GetSrcEntity()
+		dst := props.GetDstEntity()
+
+		if src.GetId() != deploymentID && dst.GetId() != deploymentID {
+			return false
+		}
+
+		if !networkgraph.AnyExternal(src, dst) || networkgraph.AllExternal(src, dst) {
+			// only looking for external flows, and not external pairs
+			return false
+		}
+
+		if !networkgraph.AnyExternalInFilter(src, dst, entityFilter) {
+			return false
+		}
+
+		return true
+	}
+
+	flowStore, err := m.getFlowStore(ctx, deployment.GetClusterId())
+	if err != nil {
+		return nil, err
+	}
+
+	flows, _, err := flowStore.GetMatchingFlows(ctx, pred, since)
+	if err != nil {
+		return nil, err
+	}
+
+	peers := make([]*v1.NetworkBaselineStatusPeer, 0, len(flows))
+
+	for _, flow := range flows {
+		peers = append(peers, m.mapFlowToPeer(flow, entitiesMap))
+	}
+
+	return peers, nil
+}
+
+func (m *manager) mapFlowToPeer(flow *storage.NetworkFlow, entitiesMap map[string]*storage.NetworkEntityInfo) *v1.NetworkBaselineStatusPeer {
+	var entity *v1.NetworkBaselinePeerEntity
+	ingress := false
+
+	props := flow.GetProps()
+	src, dst := props.GetSrcEntity(), props.GetDstEntity()
+
+	if src.GetType() == storage.NetworkEntityInfo_DEPLOYMENT {
+		info := entitiesMap[dst.GetId()]
+		entity = &v1.NetworkBaselinePeerEntity{
+			Id:         dst.GetId(),
+			Type:       dst.GetType(),
+			Name:       info.GetExternalSource().GetName(),
+			Discovered: info.GetExternalSource().GetDiscovered(),
+		}
+	} else {
+		info := entitiesMap[src.GetId()]
+		entity = &v1.NetworkBaselinePeerEntity{
+			Id:         src.GetId(),
+			Type:       src.GetType(),
+			Name:       info.GetExternalSource().GetName(),
+			Discovered: info.GetExternalSource().GetDiscovered(),
+		}
+		ingress = true
+	}
+
+	return &v1.NetworkBaselineStatusPeer{
+		Entity:   entity,
+		Port:     props.GetDstPort(),
+		Protocol: props.GetL4Protocol(),
+		Ingress:  ingress,
+	}
+}
+
+func (m *manager) getEntitiesByQuery(ctx context.Context, clusterId, query string) ([]*storage.NetworkEntity, error) {
+	q, err := search.ParseQuery(query, search.MatchAllIfEmpty())
+	if err != nil {
+		return nil, errors.Wrap(errox.InvalidArgs, err.Error())
+	}
+
+	// Retrieves entities where the cluster ID matches the request cluster OR where the cluster ID is empty indicating global entities.
+	clusterMatch := search.DisjunctionQuery(
+		search.MatchFieldQuery(search.ClusterID.String(), search.ExactMatchString(""), false),
```

Comment:
```suggestion
		search.MatchFieldQuery(search.ClusterID.String(), search.NewQueryBuilder().
              AddNullField(search.ClusterID.String()).ProtoQuery(), false),
```

I think `search.ExactMatchString("")` might be problematic. I think I have tried this before and it doesn't work. I think this is used in one other place as well and I think it should be changed there too. I am not 100% if my suggested change there is correct. Postgresql unit tests would be helpful in determining if it is.

See the conversation here https://github.com/stackrox/stackrox/pull/14319#discussion_r1959484393

Reply:
Note that the conversation says that `AddNullField` doesn't work, but due to changes in that query it should work now.

Reply:
It looks like the only place that `search.ExactMatchString("")` is used currently is the link that you provided

```
$ git grep 'search.ExactMatchString("")'
central/networkgraph/service/service_impl.go:           search.MatchFieldQuery(search.ClusterID.String(), search.ExactMatchString(""), false),
```

Look in `pkg/search/postgres/queryuuid_query.go`. You will see the following
```
        if value == pkgSearch.NullString {
                return WhereClause{
                        Query:  fmt.Sprintf("%s is null", columnName),
                        Values: []interface{}{},
                        equivalentGoFunc: func(foundValue interface{}) bool {
                                foundVal := strings.ToLower(foundValue.(string))
                                return foundVal == ""
                        },
                }, nil
        }
```
Thus even though the `-` string is used, it correctly searches for when the field is null.

I think the other place where `search.ExactMatchString("")` is used is also a bug. Please write postgres unit tests for these two functions that exercise the use of `search.ExactMatchString("")`.

Reply:
I've made this change here, and will fix the other occurrence in a follow up PR